environments/lcls/__init__.py

```diff
@@ -162,92 +168,11 @@ def set_variables(self, variable_inputs: Dict[str, float]):
-        logging.info(f'Get value of {points} points')
-
-        # Sleep for a while to get enough data
-        try:
-            rate = self.interface.get_value('EVNT:SYS0:1:LCLSBEAMRATE')
-            logging.info(f'Beam rate: {rate}')
-            nap_time = points / (rate * 1.0)
-        except Exception as e:
-            nap_time = 1
-            logging.warn(
-                'Something went wrong with the beam rate calculation. Let\'s sleep 1 second.')
-            logging.warn(f'Exception was: {e}')
-
-        time.sleep(nap_time)
-
-        if hxr:
-            PV_gas = f'GDET:FEE1:{self.fel_channel}:ENRCHSTCUHBR'
-        else:  # SXR
-            PV_gas = 'EM1K0:GMD:HPS:milliJoulesPerPulseHSTCUSBR'
-        PV_loss = self.loss_pv
-        try:
-            results_dict = self.interface.get_values([PV_gas, PV_loss])
-            intensity_raw = results_dict[PV_gas][-points:]
-            loss_raw = results_dict[PV_loss][-points:]
-            ind_valid = ~np.logical_or(np.isnan(intensity_raw), np.isnan(loss_raw))
-            intensity_valid = intensity_raw[ind_valid]
-            loss_valid = loss_raw[ind_valid]
-
-            gas_p80 = percent_80(intensity_valid)
-            gas_mean = np.mean(intensity_valid)
-            gas_median = np.median(intensity_valid)
-            gas_std = np.std(intensity_valid)
-
-            loss_p80 = percent_80(loss_valid)
-
-            return gas_p80, gas_mean, gas_median, gas_std, loss_p80
-        except Exception:  # if average fails use the scalar input
-            if hxr:  # we don't have scalar input for HXR
-                raise BadgerEnvObsError
-            else:
-                gas = self.interface.get_value('EM1K0:GMD:HPS:milliJoulesPerPulse')
-
-                return gas, gas, gas, 0, 0
-
-    def get_loss(self):  # if only loss is observed
-        points = self.points
-        logging.info(f'Get value of {points} points')
-
-        try:
-            rate = self.interface.get_value('EVNT:SYS0:1:LCLSBEAMRATE')
-            logging.info(f'Beam rate: {rate}')
-            nap_time = points / (rate * 1.0)
-        except Exception as e:
-            nap_time = 1
-            logging.warn(
-                'Something went wrong with the beam rate calculation. Let\'s sleep 1 second.')
-            logging.warn(f'Exception was: {e}')
-
-        time.sleep(nap_time)
-
-        PV_loss = self.loss_pv
-        try:
-            loss_raw = self.interface.get_value(PV_loss)[-points:]
-            ind_valid = ~np.isnan(loss_raw)
-            loss_valid = loss_raw[ind_valid]
-            loss_p80 = percent_80(loss_valid)
-
-            return loss_p80
-        except Exception:  # we don't have scalar input for loss
-            raise BadgerEnvObsError
-
-    def is_pulse_intensity_observed(self, observable_names):
-        return len([name for name in observable_names if
-                    name.startswith('pulse_intensity')])
-
     def is_beam_loss_observed(self, observable_names):
```

Comment:
Since you put `is_pulse_intensity_observed` into the FEL module you probably also want to put `is_beam_loss_observed` there too

Reply:
But I'm not sure those methods belong to the FEL module though, I feel they are more related to the specific env, like, we can of course change the name of observable `pulse intensity` from 'pulse_intensity' to 'fel_signal' or something. Then the way to determine if the user is trying to get `pulse intensity` has to change accordingly (that's why this method is more coupled to the env def, not the underlying FEL stuff) but the actual way of getting that observable remains unchanged (so they should go into the FEL module, as you already did)